src/server/cea_608_encoder/byte_pair_generator.py

```diff
@@ -7,6 +11,21 @@
     'CEA-608'
 ]
 
+def save_json_to_file(caption_data: dict):
+    # datetime object containing current date and time
+    now = datetime.now()
+    # mm.dd.YY_H:M:S
+    dt_string = now.strftime("%m.%d.%Y_%H:%M:%S")
+
+    path = pathlib.Path(__file__).parent.parent.parent.parent
+    path = str(path) + '/data/byte_pairs/'
+    file_name = 'output_' + dt_string + '.json'
+    try:
+        with open(path + file_name, 'w', encoding='utf-8') as file:
+            json.dump(caption_data, file, ensure_ascii=False, indent=4)
+    except IOError as err:
+        app.logger.error(f'Could not write JSON to file: {err}')
```

Comment:
Our `app` is not defined here this will never write out the error

Reply:
fixed